### PR TITLE
gpios: Add multi-letter bank support

### PR DESCRIPTION
**Pull request by amboar**
Signed-off-by: Andrew Jeffery <andrew@aj.id.au>

### CODE REVIEW

File: num-to-gpio [hunk 1]
```diff
@@ -2,8 +2,8 @@
 # Copyright 2016 Joel Stanley <joel@jms.id.au>
 # Apache 2.0
 
-def num_to_gpio(num):
-    num = int(num) % 320
+def num_to_gpio(base, num):
+    num = int(num) - int(base)
```
Comment (shenki) [inline num-to-gpio:6]: Before this worked with the base-added or base omitted. Can we still support that?

**Comment by shenki** [comment]
Merged after moving back to mod instead of minus

**Comment by amboar** [comment]
Eh? 320 isn't the number of GPIOs, so mod doesn't make sense. Mod op will break if we have more than 256 GPIOs (we already have 232, but this is a little academic at the moment) or the top of the GPIO space (512) drops by 24 or more for whatever reason (probably also academic), but the point is mod doesn't seem like the right operation to me

**Comment by shenki** [comment]
It's the base, not the number of gpios. The way the tool works is you can do this:
```
$ num-to-gpio 320
GPIOA0
```
Or this:
```
$ num-to-gpio 0
GPIOA0
```
So doing mod makes it work with supplying the base:
```
$ num-to-gpio 320 320
GPIOA0
$ num-to-gpio 320 0
GPIOA0
```
It also works with other bases. You're right in that it will break when we have more GPIOs than the base number, but we don't have that yet.

**Comment by amboar** [comment]
Yes, I'm aware of what 320 represents, and how mod and subtract work. Subtract works just as well without the future limitations potentially imposed by mod, but  ¯\_(ツ)_/¯ it's hard to support both without supplying at least one extra piece of information (eg base)

Thanks for merging!

**Pull request closed without merging**